checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java

```diff
@@ -252,10 +267,24 @@ public Void visitMemberSelect(
         }
 
         /**
-         * Reports an error if {@code node} represents explicitly constructing a {@code @Det
-         * HashSet}. If {@code Det} wasn't explicitly written, but the constructor would resolve to
-         * {@code @Det}, inserts {@code @OrderNonDet} instead. Also reports an error if the result
-         * of the constructor would resolve to any variant of {@code @PolyDet}.
+         * Reports an error if {@code node} represents explicitly constructing a
+         *
+         * <ol>
+         *   <li>{@code @Det HashSet}
+         *   <li>{@code @Det HashMap}
+         *   <li>{@code @OrderNonDet TreeSet}
```

Comment:
If the user passes a `@NonDet Comparator`, we conservatively return a `@NonDet TreeMap`. Do you think this is a problem?

Reply:
What I thought might be a problem is explicitly disabling `@OrderNonDet TreeMap` since this might be a possible type in the future. If that happens, though, we can just remove the check.